spynnaker/pyNN/models/neuron/synaptic_manager.py

```diff
@@ -592,120 +606,575 @@ def _write_synaptic_matrix_and_master_population_table(
+                # If there is a single matrix for the app vertex, jump over the
+                # matrix and any padding space
+                syn_block_addr = self.__next_app_syn_block_addr(
+                    syn_block_addr, pre_slice.n_atoms,
+                    max_row_info.undelayed_max_bytes, syn_max_addr)
+            elif app_key_info is None:
+                # If there isn't a single matrix, add master population table
+                # entries for each incoming machine vertex
+                r_info = routing_info.get_routing_info_for_edge(m_edge)
+                if is_undelayed:
+                    m_key_info = _AppKeyInfo(
+                        r_info.first_key, r_info.first_mask, 0, 0, 0)
+                    block_addr, syn_mat_offset = self.__reserve_mpop_block(
+                        block_addr, max_row_info.undelayed_max_bytes,
+                        max_row_info.undelayed_max_words, m_key_info,
+                        all_syn_block_sz, pre_slice.n_atoms,
+                        post_vertex_slice, synapse_info)
+                    key = (m_edge, synapse_info, post_vertex_slice.lo_atom)
+                    self.__m_edge_info[key] = (
+                        syn_mat_offset, max_row_info.undelayed_max_words,
+                        block_addr - syn_mat_offset, False)
+                elif r_info is not None:
+                    self.__poptable_type.add_invalid_entry(
+                        r_info.first_key_and_mask, 0, 0, 0)
+            # Do the same as the above for delay vertices too
+            if is_delayed and delay_app_key_info is not None:
+                delay_block_addr = self.__next_app_syn_block_addr(
+                    delay_block_addr,
+                    pre_slice.n_atoms * app_edge.n_delay_stages,
+                    max_row_info.delayed_max_bytes, delay_max_addr)
+            elif delay_app_key_info is None:
+                delay_key = (app_edge.pre_vertex, pre_slice.lo_atom,
+                             pre_slice.hi_atom)
+                r_info = self.__delay_key_index.get(delay_key, None)
+                if is_delayed:
+                    m_key_info = _AppKeyInfo(
+                        r_info.first_key, r_info.first_mask, 0, 0, 0)
+                    block_addr, d_mat_offset = self.__reserve_mpop_block(
+                        block_addr, max_row_info.delayed_max_bytes,
+                        max_row_info.delayed_max_words, m_key_info,
+                        all_syn_block_sz,
+                        pre_slice.n_atoms * app_edge.n_delay_stages,
+                        post_vertex_slice, synapse_info)
+                    key = (m_edge, synapse_info, post_vertex_slice.lo_atom)
+                    self.__delay_m_edge_info[key] = (
+                        d_mat_offset, max_row_info.delayed_max_words,
+                        block_addr - d_mat_offset, False)
+                elif r_info is not None:
+                    self.__poptable_type.add_invalid_entry(
```

Comment:
this feels like default params. ive seen it numerious times now

Reply:
Done